libpromises/evalfunction.c

```diff
@@ -1668,6 +1673,193 @@ static FnCallResult FnCallSplayClass(EvalContext *ctx,
 
 /*********************************************************************/
 
+#ifdef HAVE_LIBCURL
+static size_t cfengine_curl_write_callback(char *ptr, size_t size, size_t nmemb, void *userdata)
+{
+    unsigned int old = BufferSize((Buffer*) userdata);
+    BufferAppend((Buffer*) userdata, ptr, size*nmemb);
+    return BufferSize((Buffer*) userdata) - old;
+}
+#endif
+
+static FnCallResult FnCallCurl(ARG_UNUSED EvalContext *ctx,
+                               ARG_UNUSED const Policy *policy,
+                               ARG_UNUSED const FnCall *fp,
+                               const Rlist *finalargs)
+{
+#ifdef HAVE_LIBCURL
+    if (NULL == CURL_CACHE)
+    {
+        CURL_CACHE = JsonObjectCreate(10);
```

Comment:
Is there a nice way to tell valgrind to ignore the leak-on-exit implied by this ?